deduplicate.go

```diff
@@ -0,0 +1,31 @@
+package opslevel_jq_parser
+
+import (
+	"fmt"
+
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+func Deduplicated[T any](objects []T, keyFunc func(object T) string) []T {
+	out := make([]T, 0)
+	set := make(map[string]struct{})
+	for _, obj := range objects {
+		key := keyFunc(obj)
+		if _, ok := set[key]; ok {
+			continue
+		}
+		set[key] = struct{}{}
```

Comment:
Since `set` is being used as a check to (not) append to `out`, couldn't `set` just be a slice?
You could use [slices.Contains()](https://pkg.go.dev/golang.org/x/exp/slices#Contains)

Reply:
@davidbloss the problem with using `slices.Contains()` is that it would get the same result except it would use more memory and take more time with the same number of lines of code. 

While it is ugly to see `struct{}{}` it is a common thing to do in Go since there's no native support for sets and `struct{}` instances take 0 memory.